lib/std/debug.zig

```diff
@@ -424,6 +424,52 @@ pub const StackIterator = struct {
         return address;
     }
 
+    fn isValidMemory(address: usize) bool {
+        const aligned_address = address & ~@intCast(usize, (mem.page_size - 1));
+
+        // If the address does not span 2 pages, query only the first one
+        const length: usize = if (aligned_address == address) mem.page_size else 2 * mem.page_size;
```

Comment:
@m-radomski I'm trying to understand post factum, what is the argument here for checking if not only the current page but also the subsequent one are currently mapped by the OS? Put it another way, `msync` will fail should a single page out of the two we are syncing were not mapped. Are we guaranteed this will never happen? On arm64 macOS for instance the page size is a tremendous 16KB meaning we might not map two subsequent pages and as a result not printing the stack trace on encountering a trap since `msync` and by extension `isValidMemory` return error/false. Additionally, what is the point of syncing more than one page given that we will issue a call to `msync` on every iteration of the stack trace walker?

Reply:
Hi, sorry for the late response, but the logic was ported from libunwind and after looking closely I think you are right. It ought to check just the single page. I think my go-to-definition got lost and transported me to a file for loongarch64 instead of x86. For anyone that might still be interested, search for "validate_mem()" function in libunwind.